options_test.go

```diff
@@ -0,0 +1,459 @@
+			t.Log("test:", c.description)
+			t.Errorf("expected %s but got %s", expected, e)
+		}
+		if len(c.opt.closers) != c.numClosers {
+			t.Log("test:", c.description)
+			t.Errorf("expected to have %d closers but had %d", c.numClosers, len(c.opt.closers))
+		}
+		c.opt.Close()
+	}
+}
+
+func TestGetFirecrackerNetworkingConfig(t *testing.T) {
+	cases := []struct {
+		description string
+		opt         options
+		expectedErr func(error) (bool, error)
+		expectedNic []firecracker.NetworkInterface
+	}{
+		// FcNicConfig len 0
+		// expect empty/nil slice
+		{
+			description: "Empty FCNicConfig",
+			opt:         options{},
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			expectedNic: nil,
+		},
+		// FcNicConfig > 0 but parseNicConfig error
+		// expect parseNicconfigError
+		{
+			description: "non-empty but invalid FcNicConfig",
+			opt: options{
+				FcNicConfig: "invalid",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == parseNicConfigError, parseNicConfigError
+			},
+			expectedNic: nil,
+		},
+		// FcNicConfig valid NIC config with mdds set
+		// verify that list contains right number of entries (1) and that
+		// it has the right values
+		{
+			description: "valid FcNicConfig with mdds set to true",
+			opt: options{
+				FcNicConfig:   "valid/things",
+				validMetadata: 42,
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			expectedNic: []firecracker.NetworkInterface{
+				firecracker.NetworkInterface{
+					MacAddress:  "things",
+					HostDevName: "valid",
+					AllowMDDS:   true,
+				},
+			},
+		},
+		// FcNicConfig valid NIC config with mdds not set
+		// verify that list contains right number of entries (1) and that
+		// it has the right values/mdds not set to true
+		{
+			description: "valid FcNicConfig with mdds set to false",
+			opt: options{
+				FcNicConfig: "valid/things",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == nil, nil
+			},
+			expectedNic: []firecracker.NetworkInterface{
+				firecracker.NetworkInterface{
+					MacAddress:  "things",
+					HostDevName: "valid",
+					AllowMDDS:   false,
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		nic, err := c.opt.getNetwork()
+		if ok, expected := c.expectedErr(err); !ok {
+			t.Errorf("expected %s but got %s", expected, err)
+		}
+		if !reflect.DeepEqual(nic, c.expectedNic) {
+			t.Errorf("expected %v but got %v", c.expectedNic, nic)
+		}
+	}
+}
+
+func TestGetBlockDevices(t *testing.T) {
+	tempFile, err := ioutil.TempFile("", "firectl-test-drive-path")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.Remove(tempFile.Name())
+	defer tempFile.Close()
+	cases := []struct {
+		description    string
+		opt            options
+		expectedErr    func(e error) (bool, error)
+		expectedDrives []models.Drive
+	}{
+		// FcAdditionalDrives invalid
+		// expect a parsing error
+		{
+			description: "invalid FcAdditionalDrives value",
```

Comment:
This isn't used anywyhere in this test.

Reply:
Going to refactor to use your name suggestion which will serve the purpose this was intended for.